File: 백준/Silver/3896. 소수 사이 수열/소수 사이 수열.c
#include <stdio.h>
#include <stdlib.h>
#define MAX_VALUE 1299710

int T, k, prime[MAX_VALUE];

int main(void) {
	prime[1] = 1;
	for(int i=2;i<MAX_VALUE;i++) {
		if (!prime[i]) {
			for(int j=i*2;j<MAX_VALUE;j+=i) prime[j] = 1;
		}
	}
	scanf("%d", &T);

	while (T--) {
		scanf("%d", &k);
		if (!prime[k]) {
			printf("0\n");
			continue;
		}
		int left = k, right = k;
		while (prime	[left]) left--;
		while (prime[right]) right++;
		printf("%d\n", right-left);
	}
}

File: 백준/Silver/3896. 소수 사이 수열/소수 사이 수열.py
import sys
input = sys.stdin.readline

SIZE = 1299710

prime = [True for _ in range(SIZE)]
prime[1] = False

for i in range(2,SIZE):
    if prime[i]:
        for j in range(i*2,SIZE,i):
            prime[j] = False

T = int(input())

for _ in range(T):
    k = int(input())
    if prime[k]:
        print(0)
        continue
    left, right = k,k
    while not prime[left]: left-=1
    while not prime[right]: right+=1
    print(right-left)